### PR TITLE
Backport #4101 to beta

### PR DESCRIPTION
**Pull request by Manishearth**
This lint has been causing lots of problems.

I'll check up on other potential beta backports when I build the new changelog

r? @oli-obk

### CODE REVIEW

**Comment by Manishearth** [comment]
r? @oli-obk @phansch 

(if r+, just merge directly. I'll deal with upstream backporting later)

**Comment by flip1995** [comment]
The test files are broken. This was fixed in 08d2a0d6b20bc725305d0462ccbc9b2839820489. Should we fix this here or just ignore it (it doesn't really matter for the backport)?

**Comment by flip1995** [comment]
Squashed the commits, for less noice on the backport branch.

@bors r=oli-obk

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: rust
 
-rust: nightly
+rust: beta
```
Comment (phansch) [inline .travis.yml]: I think also AppVeyor will have to use beta? or are we going to ignore it for the backport?

**Comment by flip1995** [comment]
"All checks have passed" 🎉 Now it's bors turn

@bors r+

**Comment by Manishearth** [comment]
Yeah it's fine it the tests are kinda broken.

I'll put up a beta backport PR today

**Pull request merged**